### PR TITLE
Align long Aggregator implementation with Double and Float

### PR DESCRIPTION
**Pull request by nishantmonu51**
Add LongMin/Max aggregator combiners
Extract common code from LongSum/Min/MaxAggregatorFactories in
SimpleLongAggregatorFactory

### CODE REVIEW

File: processing/src/main/java/io/druid/query/aggregation/LongMaxAggregatorFactory.java [hunk 1]
```diff
@@ -22,48 +22,27 @@
...
 import java.util.Arrays;
-import java.util.Collections;
-import java.util.Comparator;
 import java.util.List;
-import java.util.Objects;
 
 /**
  */
-public class LongMaxAggregatorFactory extends AggregatorFactory
+public class LongMaxAggregatorFactory extends SimpleLongAggregatorFactory
```
Comment (leventov) [inline processing/src/main/java/io/druid/query/aggregation/LongMaxAggregatorFactory.java]: Please make `LongMaxAggregatorFactory`, `DoubleMaxAggregatorFactory` and `FloatMaxAggregatorFactory` identical except the primitive type and class name. Currently there are some pairwise differences between them: method order, constructor parameter order, the list of overridden methods, utility method usage, etc.

File: processing/src/main/java/io/druid/query/aggregation/LongMinAggregatorFactory.java [hunk 2]
```diff
@@ -22,49 +22,27 @@
...
 import java.util.Arrays;
-import java.util.Collections;
-import java.util.Comparator;
 import java.util.List;
-import java.util.Objects;
 
 /**
  */
-public class LongMinAggregatorFactory extends AggregatorFactory
+public class LongMinAggregatorFactory extends SimpleLongAggregatorFactory
```
Comment (leventov) [inline processing/src/main/java/io/druid/query/aggregation/LongMinAggregatorFactory.java]: Same ask as about `LongMaxAggregatorFactory`

File: processing/src/main/java/io/druid/query/aggregation/LongSumAggregatorFactory.java [hunk 3]
```diff
@@ -22,29 +22,18 @@
...
 import java.util.Arrays;
-import java.util.Collections;
-import java.util.Comparator;
 import java.util.List;
-import java.util.Objects;
 
 /**
  */
-public class LongSumAggregatorFactory extends AggregatorFactory
+public class LongSumAggregatorFactory extends SimpleLongAggregatorFactory
```
Comment (leventov) [inline processing/src/main/java/io/druid/query/aggregation/LongSumAggregatorFactory.java]: Same

File: processing/src/main/java/io/druid/query/aggregation/SimpleLongAggregatorFactory.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+import io.druid.math.expr.Parser;
+import io.druid.segment.BaseLongColumnValueSelector;
+import io.druid.segment.ColumnSelectorFactory;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+
+public abstract class SimpleLongAggregatorFactory extends AggregatorFactory
```
Comment (leventov) [inline processing/src/main/java/io/druid/query/aggregation/SimpleLongAggregatorFactory.java]: Same

File: processing/src/main/java/io/druid/query/aggregation/LongMaxAggregatorFactory.java (hunk 1 above)
Comment (nishantmonu51) [reply processing/src/main/java/io/druid/query/aggregation/LongMaxAggregatorFactory.java]: done.

File: processing/src/main/java/io/druid/query/aggregation/LongMinAggregatorFactory.java (hunk 2 above)
Comment (nishantmonu51) [reply processing/src/main/java/io/druid/query/aggregation/LongMinAggregatorFactory.java]: done.

File: processing/src/main/java/io/druid/query/aggregation/LongSumAggregatorFactory.java (hunk 3 above)
Comment (nishantmonu51) [reply processing/src/main/java/io/druid/query/aggregation/LongSumAggregatorFactory.java]: done.

File: processing/src/main/java/io/druid/query/aggregation/SimpleLongAggregatorFactory.java (hunk 4 above)
Comment (nishantmonu51) [reply processing/src/main/java/io/druid/query/aggregation/SimpleLongAggregatorFactory.java]: done.

**Comment by leventov** [comment]
@nishantmonu51 please also rename: https://github.com/druid-io/druid/pull/5861#pullrequestreview-127200074

**Comment by nishantmonu51** [comment]
@leventov done.

**Pull request merged**